test/staking/StakingRewards.test.ts

```diff
@@ -0,0 +1,217 @@
+      "1000000" //100% rr
+    );
+    cDAI1 = await cdaiFactory.deploy(dai.address);
+    const cdaiLowWorthFactory = await ethers.getContractFactory(
+      "cDAILowWorthMock"
+    );
+    cDAI2 = await cdaiLowWorthFactory.deploy(dai.address);
+    const cdaiNonMintableFactory = await ethers.getContractFactory(
+      "cDAINonMintableMock"
+    );
+    cDAI3 = await cdaiNonMintableFactory.deploy(dai.address);
+    await marketMaker.initializeToken(
+      cDAI1.address,
+      "100", //1gd
+      "10000", //0.0001 cDai
+      "1000000" //100% rr
+    );
+    await marketMaker.initializeToken(
+      cDAI2.address,
+      "100", //1gd
+      "10000", //0.0001 cDai
+      "1000000" //100% rr
+    );
+    await marketMaker.initializeToken(
+      cDAI3.address,
+      "100", //1gd
+      "10000", //0.0001 cDai
+      "1000000" //100% rr
+    );
+    await marketMaker.transferOwnership(goodReserve.address);
+
+    setDAOAddress("CDAI", cDAI.address);
+    setDAOAddress("DAI", dai.address);
+
+    const ictrl = await ethers.getContractAt(
+      "Controller",
+      controller,
+      schemeMock
+    );
+
+    //This set addresses should be another function because when we put this initialization of addresses in initializer then nameservice is not ready yet so no proper addresses
+    await goodReserve.setAddresses();
+    //Set  Goodfundmanager's reserve
+    const encodedData = goodFundManagerFactory.interface.encodeFunctionData(
+      "setReserve",
+      [goodReserve.address]
+    );
+    const encodedDataTwo = goodFundManagerFactory.interface.encodeFunctionData(
+      "setStakingReward",
+      ["1000", goodCompoundStaking.address,"0","100"] // set 10 gd per block
+    );
+    await ictrl.genericCall(goodFundManager.address, encodedData, avatar, 0);
+    await ictrl.genericCall(goodFundManager.address, encodedDataTwo, avatar, 0);
+    await setDAOAddress("MARKET_MAKER", marketMaker.address);
+    await setDAOAddress("FUND_MANAGER", goodFundManager.address);
+  });
+
+  it("should get g$ minting permissions", async () => {
+    expect(await goodReserve.dao()).to.be.equal(controller);
+    expect(await goodReserve.avatar()).to.be.equal(avatar);
+    await goodReserve.start();
+  });
+
+  it("should be set fundmanager address in NameService", async () => {
+    let fundManagerAddress = await nameService.getAddress("FUND_MANAGER");
+    expect(fundManagerAddress).to.be.equal(goodFundManager.address);
+  });
+
+  it("should be set rewards per block for particular stacking contract", async () => {
+    let rewardPerBlock= await goodFundManager.getStakingReward(
+      goodCompoundStaking.address
+    );
+    expect(rewardPerBlock[0].toString()).to.be.equal("1000");
+    expect(rewardPerBlock[1].toString()).to.be.equal("0")
+    expect(rewardPerBlock[2].toString()).to.be.equal("100")
+    expect(rewardPerBlock[3]).to.be.equal(false)
+    
+  });
+
+  it("should be able to earn rewards after some block passed", async () => {
```

Comment:
we need much more tests here to check all aspects of our rewards calculation
you can also copy tests from baseshare etc

Reply:
I'll add more when cancel out old interest mechanism after #39